addons/mail/models/res_partner.py

```diff
@@ -45,11 +45,7 @@ def _get_needaction_count(self):
     def _get_starred_count(self):
         """ compute the number of starred of the current partner """
         self.ensure_one()
-        self.env.cr.execute("""
-            SELECT count(*) as starred_count
-            FROM mail_message_res_partner_starred_rel R
-            WHERE R.res_partner_id = %s """, (self.id,))
-        return self.env.cr.dictfetchall()[0].get('starred_count')
+        return self.env['mail.message'].with_user(self.user_ids[0]).search_count([('starred', '=', True)])
```

Comment:
Why not simply ``return self.env['mail.message'].search_count([('starred_partner_ids', 'in', self.ids)])`` ? as it does the same anyway. Obviously this depends on current user's ACLs but that seems legit (you should not have info on records you cannot see).

Reply:
As I saw while testing, using `return self.env['mail.message'].search_count([('starred_partner_ids', 'in', self.ids)])` breaks `test_mail_message_starred_private_channel` because a starred message from a private channel is still considered even if we don't have access to this channel anymore

Reply:
The current flow of calling this method is `'starred_counter': self.partner_id._get_starred_count(),` where `self` is user. It makes no sense to lose the user context that we had just before by having the method on partner and then guess at this step which user should be used... -> If we need the user on the method for any reason, it should be a user method instead. (you can keep this one here in stable just for stable policy)

Or if you can remove the `with_user` it would be ok too, about this test, I'm not sure if we actually care to keep a starred message that became inaccessible ? Probably not...

Reply:
Why not removing the call to ``_get_starred_count`` and implement what you need in ``_init_messaging`` then ? We are adding layers on top of layers, code is becoming impossible to understand ... easier to be on the point. And this method can gain a docstring telling "it may be wrong bisous".